geopandas/io/arrow.py

```diff
@@ -94,7 +94,25 @@ def _create_metadata(df, schema_version=None):
     column_metadata = {}
     for col in df.columns[df.dtypes == "geometry"]:
         series = df[col]
-        geometry_types = sorted(Series(series.geom_type.unique()).dropna())
+
+        # Defines geometry_types according to the geoparquet standard:
+        # https://github.com/opengeospatial/geoparquet/blob/main/format-specs/geoparquet.md#geometry_types
+        # Instead of going through the `_vectorized.geometry_type_values`
+        # that do not include the "Z" varieties.
+        geometry_types = sorted(
+            Series(
+                list(
+                    set(
+                        [
+                            f"{geom} Z" if has_z else geom
+                            for geom, has_z in zip(series.geom_type, series.has_z)
```

Comment:
Could you maybe write this in two separate steps? (eg first creating the set, and then the `sorted(Series(list(..` on a next line) That should help with the formatting here

Reply:
Sorry for the late reply! Ok this is easily fixed, I am unsure on how to proceed with the testing. Looking into it!

Reply:
Fixed in the last commit